test/Mono.Linker.Tests.Cases/DataFlow/MethodOutParameterDataFlow.cs

```diff
@@ -66,35 +66,35 @@ static void TestInitializedReadFromOutParameter_MismatchOnOutput ()
 
 		// https://github.com/dotnet/linker/issues/2632
 		// This test should generate a warning since there's mismatch on annotations
-		// [ExpectedWarning ("IL2062", nameof (DataFlowTypeExtensions.RequiresPublicFields))]
+		[ExpectedWarning ("IL2118", nameof (DataFlowTypeExtensions.RequiresPublicFields))]
 		static void TestInitializedReadFromOutParameter_MismatchOnOutput_PassedTwice ()
 		{
 			Type typeWithMethods = null;
 			TryGetAnnotatedValueFromValue (typeWithMethods, out typeWithMethods);
 			typeWithMethods.RequiresPublicFields ();
 		}
 
-		[ExpectedWarning ("IL2072", nameof (TryGetAnnotatedValue))]
 		// https://github.com/dotnet/linker/issues/2632
-		// This second warning should not be generated, the value of typeWithMethods should have PublicMethods
+		// This warning should not be generated, the value of typeWithMethods should have PublicMethods
 		// after the call with out parameter.
-		[ExpectedWarning ("IL2072", nameof (DataFlowTypeExtensions.RequiresPublicMethods))]
+		[ExpectedWarning ("IL2072", nameof (DataFlowTypeExtensions.RequiresPublicMethods), ProducedBy = ProducedBy.Analyzer)]
 		static void TestInitializedReadFromOutParameter_MismatchOnInput ()
 		{
 			Type typeWithMethods = GetTypeWithFields ();
+			// No warning on out parameter
 			TryGetAnnotatedValue (out typeWithMethods);
 			typeWithMethods.RequiresPublicMethods ();
 		}
 
-		[ExpectedWarning ("IL2072", nameof (TryGetAnnotatedValueFromValue))]
 		[ExpectedWarning ("IL2072", nameof (TryGetAnnotatedValueFromValue))]
 		// https://github.com/dotnet/linker/issues/2632
-		// This third warning should not be generated, the value of typeWithMethods should have PublicMethods
+		// This warning should not be generated, the value of typeWithMethods should have PublicMethods
 		// after the call with out parameter.
-		[ExpectedWarning ("IL2072", nameof (DataFlowTypeExtensions.RequiresPublicMethods))]
+		[ExpectedWarning ("IL2072", nameof (DataFlowTypeExtensions.RequiresPublicMethods), ProducedBy = ProducedBy.Analyzer)]
```

Comment:
Do we have an issue tracking the fact that analyzer doesn't seem to handle ref parameters correctly?